.github/workflows/release.yaml

```diff
@@ -21,36 +34,72 @@ jobs:
         uses: actions/setup-node@v4
         with:
           node-version: 20.18.0
-          cache: 'pnpm'
+          cache: "pnpm"
 
       - name: Install dependencies
         run: pnpm install
 
       - name: Build
         run: pnpm build
 
-      - name: Create Draft Release
-        run: gh release create ${{ github.ref }} --draft --title "Release ${{ github.ref }}" --notes-from-tag
-
-      - name: Enter desktop app directory
-        run: cd apps/desktop
-
       - name: Import GPG key
         uses: crazy-max/ghaction-import-gpg@v6
         with:
           gpg_private_key: ${{ secrets.GPG_PRIVATE_KEY }}
           passphrase: ${{ secrets.GPG_PASSPHRASE }}
-          fingerprint: ${{secrets.GPG_KEY_ID }}
+          fingerprint: ${{ secrets.GPG_KEY_ID }}
           git_user_signingkey: true
 
       - name: Generate deb and rpm files
+        working-directory: apps/desktop
+        env:
+          GH_TOKEN: ${{ secrets.PERSONAL_GH_TOKEN }}
         run: pnpm run electron:package:linux
 
       - name: Sign the deb and rpm files using dpkg-sig and rpm-sign
+        working-directory: apps/desktop
         run: |
           sudo apt install -y dpkg-sig
           dpkg-sig --sign builder -k ${{ secrets.GPG_KEY_ID }} ./dist/*.deb
           rpm --addsign ./dist/*.rpm
 
       - name: Upload deb and rpm files
-        run: gh release upload ${{ github.ref }} ./dist/*.deb ./dist/*.rpm
+        working-directory: apps/desktop
+        env:
+          GH_TOKEN: ${{ secrets.PERSONAL_GH_TOKEN }}
+        run: gh release upload ${{  github.ref_name }} ./dist/*.deb ./dist/*.rpm
+
+  create-release-mac:
+    runs-on: macos-latest
+    needs: create-draft-release
+    environment: "Production – umami-v2"
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v4
+
+      - uses: pnpm/action-setup@v3
+        with:
+          version: 9.9.0
+
+      - name: Setup Node.js environment
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20.18.0
+          cache: "pnpm"
+
+      - name: Install dependencies
+        run: pnpm install
+
+      - name: Build
+        run: pnpm build
```

Comment:
you're building all the projects which is redundant. it's better to run turbo build with `working-directory: apps/desktop`

Reply:
Yes. Thats right.

Reply:
It does not work. Need to build the dependency projects also. need to run it from root directory.